lib/router/src/utils.test.ts

```diff
@@ -178,6 +188,26 @@ describe('buildArgsParam', () => {
     expect(param).toEqual('foo.bar[0].key:!null;foo.baz:!null');
   });
 
+  it('encodes hex color values by replacing # with !', () => {
+    const param = buildArgsParam({}, { key: '#ff4785' });
+    expect(param).toEqual('key:!ff4785');
```

Comment:
I feel like you might regret this format (to general, will limit future options). Do you think maybe it's better to do something more formal/deterministic like: `!hex(ff4785)`

Reply:
Yeah sounds good!